client/components/advanced-credentials/index.tsx

```diff
@@ -89,17 +90,32 @@ const AdvancedCredentials: FunctionComponent< Props > = ( { action, host, role }
 	const { protocol } = credentials;
 	const isAtomic = hasCredentials && 'dynamic-ssh' === protocol;
 
+	const [ testCredentialsResult, setTestCredentialsResult ] = useState( false );
+	const [ testCredentialsLoading, setTestCredentialsLoading ] = useState( true );
+
+	useEffect( () => {
+		( async () => {
+			const results: { ok: boolean } = await wp.req.post( {
+				path: '/sites/' + siteId + '/rewind/credentials/test?role=main',
+				apiNamespace: 'wpcom/v2',
+			} );
+			const { ok } = results;
+			setTestCredentialsLoading( false );
+			setTestCredentialsResult( ok );
+		} )();
+	}, [] );
+
 	const statusState = useMemo( (): StatusState => {
-		if ( isRequestingCredentials ) {
+		if ( isRequestingCredentials || testCredentialsLoading ) {
 			return StatusState.Loading;
 		}
 
-		if ( hasCredentials ) {
+		if ( hasCredentials && testCredentialsResult ) {
```

Comment:
Should we put `testCredentialsResult` in the `hasCredentials ` constant definition? Because right now someone could accidentally use `hasCredentials` and miss `testCredentialsResult`.

Reply:
Thank you for the suggestion! @jwebbdev made #62607 earlier this year, which uses `hasCredentials` to test if a site is Atomic. I've put `testCredentialsResult` in the `hasCredentials` constant definition, and in my testing, I haven't seen a conflict with regards to Atomic detection. I wonder if for any reason `testCredentialsResult` returns false, it may affect Atomic detection. John, do you have any thoughts? I wonder if perhaps we should leave this line as is, or create a new const definition here.

Reply:
Yeah, the referenced isAtomic check wouldn't benefit from a connection test being run.  It is a more lightweight check to see the type of credentials that are saved.  

Because the credentials test isn't exactly light weight (it tests a connection to the customer's site over FTP/SSH) and currently doesn't have caching implemented we probably only want to call it explicitly.  Potentially a `hasValidCredentials` selector could do it, but we might want some caching in place before it's used too widely.